### DISCUSSION TITLE
Good work

### DISCUSSION THREAD

**Discussion by Maurycy-Krzeminski** [nvim-java/nvim-java#84] [General]
Hey!
I would like to thank you for awesome neovim extension for java.
Dap is working really well with Spring Boot when I was testing it today.
However there is some problem with Quarkus.
Debuging works only in public static void main function in @QuarkusMain.
Do you think about supporting Quarkus in future, or maybe I just have it configured wrong?

**Comment by s1n7ax** [comment]
In java, there always should be a main method to start the execution. I've never heard of Quarkus. It's most likely that similar to tests, there is a executor application to Quarkus if `@QuarkusMain` just an annotation you put on a class or something.

This definitely doesn't work without a main method. I couldn't really tell if you can simply configure it without knowing how it works. Were you able to get it working with nvim-jdtls or something?

**Reply by Maurycy-Krzeminski** [reply]
With nvim-jdtls dap is working in Quarkus (https://quarkus.io/).
And today it started working also with your extension nvim-java.
I have no idea why it started working while i didn't change anything.
If i have to guess magic of developing on windows :).

![image](https://github.com/nvim-java/nvim-java/assets/13149328/d87e0ae7-5b01-43af-b14d-94b1cf34ac43)
